src/acl/FilledChecklist.h

```diff
@@ -55,14 +61,22 @@ class ACLFilledChecklist: public ACLChecklist
     /// set the client side FD
     void fd(int aDescriptor);
 
+    /// response added by updateReply()
+    /// \prec hasReply()
+    const HttpReply &reply() const { return *reply_; }
+
+    /// Remembers the given response (if it is not nil) or does nothing
+    /// (otherwise).
```

Comment:
This call is implicitly part of reference counting and should clear anything already held when receiving a nil pointer. Otherwise we risk re-using checklists with wrong transaction pointers.